client/state/activity-log/log/reducer.js

```diff
@@ -9,33 +9,31 @@ import { mapValues } from 'lodash';
  */
 import ActivityQueryManager from 'lib/query-manager/activity';
 import { ACTIVITY_LOG_UPDATE, DESERIALIZE, SERIALIZE } from 'state/action-types';
-import { createReducer, isValidStateWithSchema } from 'state/utils';
+import { isValidStateWithSchema, keyedReducer } from 'state/utils';
 import { logItemsSchema } from './schema';
 
-export const logItems = createReducer(
-	{},
-	{
-		[ ACTIVITY_LOG_UPDATE ]: ( state, { data, found, query, siteId } ) => {
-			return state[ siteId ]
-				? {
-						...state,
-						[ siteId ]: state[ siteId ].receive( data, { found, query } ),
-					}
-				: {
-						...state,
-						[ siteId ]: new ActivityQueryManager().receive( data, { found, query } ),
-					};
-		},
-		[ DESERIALIZE ]: state => {
-			if ( ! isValidStateWithSchema( state, logItemsSchema ) ) {
-				return {};
-			}
-			return mapValues( state, ( { data, options } ) => {
-				return new ActivityQueryManager( data, options );
-			} );
-		},
-		[ SERIALIZE ]: state => {
-			return mapValues( state, ( { data, options } ) => ( { data, options } ) );
-		},
+const logItem = ( state = undefined, { type, data, found, query } ) => {
+	switch ( type ) {
+		case ACTIVITY_LOG_UPDATE:
+			return ( state || new ActivityQueryManager() ).receive( data, { found, query } );
+
+		case DESERIALIZE:
+			return isValidStateWithSchema( state, logItemsSchema )
+				? new ActivityQueryManager().receive( state.data, state.options )
+				: undefined;
+
+		case SERIALIZE:
+			return { data: state.data, options: state.options };
 	}
-);
+
+	return state;
+};
+
+const logItemsReducer = keyedReducer( 'siteId', logItem );
+
+export const logItems = ( state, action ) =>
+	action.type === DESERIALIZE || action.type === SERIALIZE
+		? logItemsReducer( mapValues( state, item => logItem( item, action ) ), action )
+		: logItemsReducer( state, action );
+
+logItems.hasCustomPersistence = true;
```

Comment:
@sirreal found the problem! it was because the schema was looking for an object whose keys were site ids and whose values were state objects. I have updated the schema to operate on the _item_ and not the collection and it's working fine now.